### ISSUE TITLE
Vertical velocity estimates and ALE remapping

### ISSUE THREAD

**Issue by hmkhatri** [NOAA-GFDL/MOM6#1112]
We are looking into the layer-wise momentum budget in MOM6 in z* coordinate. All terms in the momentum budget are available as diagnostics in MOM6 except the dia-surface velocity acceleration term. Is it possible to compute this term online in the remapping step? 
Following the same, can we estimate vertical velocity online and output as a diagnostic in MOM6?

**Comment by ashao** [comment]
`wmo` is currently calculated offline as the convergence of `umo` and `vmo` cumulatively summed from the bottom up. While we had originally sanity checked that the vertically summed `wmo` matched PmE, we did not include a term that accounts for the stretching of the z* coordinate due to changes in SSH. Additionally, it is possible that we are making a systematic error because the averages of any nonlinear terms are assumed to be zero.

The implementation of this diagnostic online is non-trivial because the calculation of `wmo` depends on the remapped values of `umo` and `wmo` which are only calculated during `post_data` and unavailable afterwords. How to implement this will require some discussion on how to extend the diagnostics framework to account for these dependencies.

**Comment by StephenGriffies** [comment]
@hmkhatri asks two questions.  @ashao answered the second.

The remapping term is the first question. Does anyone have a sense for whether we can get the remapping term diagnosed online for the velocity equation, in a manner similar to how we now get the analogous term for the tracer budgets?

**Comment by ashao** [comment]
I have no solution, but more questions to ask.
First though a clarification:
>the analogous term for the tracer budgets?
@StephenGriffies you're referring to the vertical remapping tendency for a tracer `C` diagnosed as:
```
C_vert_remap_tendency = C_post_ALE*h_post_ALE - C_pre_ALE*h_pre_ALE
```
and @hmkhatri and @StephenGriffies are wondering what an analogous term is for an implied vertical velocity.

I almost feel like this is a broader question about how to transform between the vertical momentum equation in the Leclair and Madec flavor of ALE, vs. the regrid/remap flavor.

My gut reaction is that it will be hard to define this transformation because of the vertically non-local nature of the regrid, remap operation. For example. Let's say that you went from the situation 
![diagram](https://user-images.githubusercontent.com/3845639/84530277-b67e6f80-ac97-11ea-9d29-6109ba7c0233.png)
where at `t=1`, the `k=1,2` layers have vanished due to the regridding. For a tracer, this is an easy tendency to calculate as given by the above equation- the entire tracer content in the vanished layers has been depleted. 

The velocity *of* an interface can be calculated as the difference in its position pre and post ALE, but I don't know how you would define the velocity *across* the interface let alone its acceleration.

@hmkhatri @StephenGriffies , what are your thoughts on what the accelerations would be in this case?

**Comment by adcroft** [comment]
N to M layer remapping is implemented in the code by using max(N,M)+1 interfaces and implies abs(N-M) vanished layers in one or the other.

**Comment by ashao** [comment]
The question wasn't about the N-M mappings, but more about the vertical velocities that's implied by the vanishing of the bottom layers.

w velocity of the second interface is `[h(k=2)+h(k=3)]/Delta T` IF it's calculated from a fixed frame of reference. My question is more what does that means though from the frame of reference of the interface itself and moreover in the interpretation of the momentum budget diagnostic that @hmkhatri is asking about.

**Comment by adcroft** [comment]
Deep down in the algorithm the remapping calculates the change in interface position, which is your w*time step. That's why we need that same number of interfaces in both pictures and represent a change in degrees of freedom with vanished layers. The particular solution for your setup is w(1)=0, w(2)=2, w(3)=1, w(4)=0 which takes your 3 cells of thickness 1 and produces h(1)=3, h(2)=h(3)=0.

**Comment by ashao** [comment]
Gotcha, thanks for clarifying. You're referring to the `dzInterface` that comes from the grid generator, e.g. 
https://github.com/NOAA-GFDL/MOM6/blob/2179c8813e21811006f96fade03850a8ad09faf1/src/ALE/MOM_regridding.F90#L855-L861

**Comment by ashao** [comment]
So hypothetically, this is already there (without the division by Delta T):

https://github.com/NOAA-GFDL/MOM6/blob/2179c8813e21811006f96fade03850a8ad09faf1/src/ALE/MOM_ALE.F90#L397

**Comment by adcroft** [comment]
Yes, and the tendency due to remapping for scalars is https://github.com/NOAA-GFDL/MOM6/blob/2179c8813e21811006f96fade03850a8ad09faf1/src/ALE/MOM_ALE.F90#L838-L855, but a few lines down there is no similar diagnostic for the components of momentum.

**Comment by ashao** [comment]
I guess it's kind of redundant to have the `remap_cont_2d` because it should be 0.

**Comment by hmkhatri** [comment]
@ashao @adcroft  I couldn't follow much of the conversation above but it seems to me that acceleration term due to dia-surface velocity can be obtained as a diagnostic to close the momentum budget. Is that correct?

**Comment by adcroft** [comment]
Yes, a diagnostic needs to be added as mentioned in https://github.com/NOAA-GFDL/MOM6/issues/1112#issuecomment-643488958

**Comment by StephenGriffies** [comment]
@hmkhatri please give see if you can write the diagnostic and see if it makes sense.  This is encouraging.

**Comment by Biswamoy** [comment]
Hello,
I was trying to estimate vertical velocity in MOM6 with a Hybrid ALE coordinate to analyze the MOC in the Indian Ocean. Hence, I came across this post. 
Is there any update on estimating vertical velocity online or offline?
